bookwyrm/views/rss_feed.py

```diff
@@ -42,3 +42,114 @@ def items(self, obj):
     def item_link(self, item):
         """link to the status"""
         return item.local_path
+
+
+class RssReviewsOnlyFeed(Feed):
+    """serialize user's reviews in rss feed"""
+
+    description_template = "rss/content.html"
+
+    def item_title(self, item):
+        """render the item title"""
+        if hasattr(item, "pure_name") and item.pure_name:
+            return item.pure_name
+        title_template = get_template("snippets/status/header_content.html")
+        title = title_template.render({"status": item})
+        template = get_template("rss/title.html")
+        return template.render({"user": item.user, "item_title": title}).strip()
+
+    def get_object(self, request, username):  # pylint: disable=arguments-differ
+        """the user who's posts get serialized"""
+        return get_user_from_username(request.user, username)
+
+    def link(self, obj):
+        """link to the user's profile"""
+        return obj.local_path
+
+    def title(self, obj):
+        """title of the rss feed entry"""
+        return _(f"Reviews from {obj.display_name}")
+
+    def items(self, obj):
+        """the user's activity feed"""
+        return obj.status_set.select_subclasses("review").filter(
```

Comment:
`select_subclasses("review")` doesn't do what one might expect here. Instead of filtering out everything that isn't a review, it returns all reviews as `Review` objects and everything else as the parent class (`Status`). This means we still get all other public items in the feed, but with only the username as the title, and the URL, without a description.

To filter this to only reviews (or quotations, etc) query the subclass directly instead of using a reverse lookup:

```python
return Review.objects.filter(
    user=obj,
    privacy__in=["public", "unlisted"],
)[:10]
```

Reply:
I just skimmed through the documentation, but yea it looks like I actually made a mistake. Strange how that one works. Well spotted, thanks! I will update it tomorrow